pkg/sources/reconciler/common/resource/container.go

```diff
@@ -173,19 +173,27 @@ func Probe(path, port string) ObjectOption {
 func StartupProbe(path, port string) ObjectOption {
 	return func(object interface{}) {
 		var sp **corev1.Probe
+		var intstrPort intstr.IntOrString
 
 		switch o := object.(type) {
 		case *corev1.Container:
 			sp = &o.StartupProbe
-		case *appsv1.Deployment:
+		case *appsv1.Deployment, *servingv1.Service:
 			sp = &firstContainer(o).StartupProbe
 		}
 
+		switch object.(type) {
+		case *corev1.Container, *appsv1.Deployment:
+			intstrPort = intstr.FromString(port)
+		case *servingv1.Service:
+			// setting port explicitly is invalid in a Knative Service
+		}
```

Comment:
Although Knative doesn't yet support startup probes, I know there are plans to make it happen in the near future so I decided to include support for it in this PR.

I'm fine deleting that code if this feels unrelated to the main purpose of the PR.